foundations/src/telemetry/log/init.rs

```diff
@@ -124,6 +125,14 @@ pub(crate) fn init(service_info: &ServiceInfo, settings: &LoggingSettings) -> Bo
     Ok(())
 }
 
+/// Opens fd 1 directly with a [`BUF_SIZE`] buffer.
+///
+/// [`io::Stdout`] is line buffered and adds unnecessary overhead for JSON output.
+fn block_buffered_stdout() -> BufWriter<File> {
```

Comment:
Technically line buffering is a case of block buffering, also "block buffering" is not very often used term in the Rust realm as far as I can tell and also having it at the start of the function name one might confuse it with a `verb` thinking that it blocks buffered stdout. So, maybe something slightly goofy, but more explicit would work better, like `stdout_writer_without_line_buffering`